inspirehep/modules/forms/validators/simple_fields.py

```diff
@@ -115,8 +128,19 @@ def duplicated_doi_validator(form, field):
     if not doi:
         return
     if current_app.config.get('PRODUCTION_MODE'):
+        user_roles = _get_current_user_roles()
+
+        # First check in default collection
         inspirehep_duplicated_validator('doi:' + doi, 'DOI')
 
+        # And in Hal collection
+        try:
+            inspirehep_duplicated_validator('doi:' + doi, 'DOI', 'HAL Hidden')
```

Comment:
`inspirehep_duplicated_validator` and [this](https://github.com/inspirehep/inspire-next/blob/49a4197747193054894273b51732efd78dd4a25b/inspirehep/modules/workflows/tasks/matching.py#L90) duplicate the same (or very similar) logic.